src/util/log.cpp

```diff
@@ -0,0 +1,23 @@
+#include "log.h"
+#include <cstdio>
+#include <cstdarg>
+
+void _smek_error_log(const char *file, u32 line, const char *message, ...) {
+    std::fprintf(stderr, YELLOW "!%s" RESET "|" BLUE "%d" RESET ": ", file, line);
+    va_list args;
+    va_start(args, message);
+    std::vfprintf(stderr, message, args);
+    va_end(args);
+    std::fprintf(stderr, "\n");
+}
+
+void _smek_info_log(const char *file, u32 line, const char *message, ...) {
```

Comment:
`info` should print to `stdout` and not `stderr`.

Reply:
I thought about this, and wondered if that really was the case. Do we honestly ever want to write to stdout? Since we're not a terminal application, and if we added a printf we might want that to go out on stdout. I was kinda split on this.

Reply:
I guess it depends on what we log. If it's not too much it probably doesn't matter but if we do verbose logging it would be nice to put info on stdout and something like warn on stderr. But that's out of scope of this pr so for now it works.